src/compiler/checker.ts

```diff
@@ -46365,6 +46366,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
             if (isIdentifier(node.name)) {
                 checkCollisionsForDeclarationName(node, node.name);
+                if (!(node.flags & (NodeFlags.Namespace | NodeFlags.GlobalAugmentation))) {

```

Comment:
This will only report errors in editor and there is no way to report these errors on command line. I think we need to handle "deprecation" version here instead to issue error

Reply:
The intent is to not report the in tsc yet, per #57913